python/dgl/nn/pytorch/network_emb.py

```diff
@@ -0,0 +1,220 @@
+"""Network Embedding NN modules"""
+# pylint: disable= invalid-name
+import torch
+from torch import nn
+from torch.nn import init
+from torch.utils.data import DataLoader
+import torch.nn.functional as F
+import numpy as np
+import tqdm
+from ...sampling import random_walk
+
+
+
+class Metapath2vec(nn.Module):
+    """
+    Metapath2vec, a scalable representation learning model for heterogeneous networks
+    from KDD’17, August <https://dl.acm.org/doi/pdf/10.1145/3097983.3098036>
+    Parameters
+    ----------
+    g :DGLHeteroGraph
+        Graph data for generating embedding
+    emb_dimension :int
+        Size of each embedding vector
+    metapath :List[str]
+        The metapath is described as a list of strings for relation edges between different
+        types of nodes. It is a  sequence of edges which denotes transition  for different
+        types of nodes at each step. Note that the node type at the start and end of the
+        metapath are commonly the same.
+    window_size :int
+        The context size which is considered for positive samples.
+    min_count :int , optional
+        The number of node counts that occur less than  min_count  value in all meta-paths
+        will be discarded before random walks. (default: 0)
+    negative_samples_size :int , optional
+        The number of negative samples need to be extracted for each positive sample.(default: 5)
+    node_repeat  :int , optional
+        The number of random walks to sample for each start node$V_1$. (default: 1)
+    nid2word  :Dict[ str, Dict[ int , str ] ], optional
+        If set, we can use model.id2word  to get the dict where the key is embedding id and
+        the value is its corresponding name.(default: None)
+    sparse  :int , optional
+        If set to True, gradients w.r.t. to the weight matrix will be sparse. (default: True)
+    """
+
+    def __init__(self,
+                 g,
+                 emb_dimension,
+                 metapath,
+                 window_size,
+                 min_count=0,
+                 negative_samples_size=5,
+                 node_repeat=1,
+                 nid2word=None,
+                 sparse=True
+                 ):
+        super(Metapath2vec, self).__init__()
```

Comment:
```suggestion
        super().__init__()
```

should be enough now